test/modules/obsagent-client/src/main.rs

```diff
@@ -120,91 +43,27 @@ fn init_logging() {
 #[tokio::main]
 async fn main() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
     init_logging();
-    let config = init_config()?;
+    let config = Arc::new(config::init_config()?);
```

Comment:
So it looks like Arc _is_ needed with the current state of the prometheus_server, where I'm calling `tokio::spawn(async move { metrics_loop(config.update_rate).await });`. However, if I store a copy of `config.update_rate` in a variable first and then use that variable in the `tokio::spawn` call, I'm able to get away with using an Rc: 
`    let ur = config.update_rate;
    tokio::spawn(async move { metrics_loop(ur).await });
`
I guess the difference is that in the second approach I'm using an `f64` in the Future, and it implements `Send`, but in the first approach I was moving the pointer into the Future, so Rc wouldn't work because it doesn't implement `Send`.

I'll go with the second approach.  Nice catch!

Reply:
After taking a closer look, the config is just a bunch of strings and f64. I think you can derive the clone trait then we won't need arc or rc. We are cloning the config the way you have said above, just doing it in a weirder way.

Reply:
Ok, I went ahead and added the derived clone trait and removed the use of Rc.